src/daos/order_dao.py

```
@@ -17,6 +17,17 @@ def write_orders(market_slug: str, datetime: datetime, orders: List[Order], test
     csv_filename = os.path.join('data', f"{datetime.strftime('%Y%m%d')}_{market_slug}_orders{test_suffix}.csv")
 
     rows = [order.asdict() for order in orders]
+    
+    # Validate that Order attributes match expected field names
+    if rows:
+        order_fields = set(rows[0].keys())
+        expected_fields = set(FIELD_NAMES)
+        if order_fields != expected_fields:
+            missing_fields = expected_fields - order_fields
+            extra_fields = order_fields - expected_fields
+            error_msg = f"Order fields mismatch. Missing: {missing_fields}, Extra: {extra_fields}"
+            logger.error(error_msg)
+            raise ValueError(error_msg)
```

```suggestion
```
Instead of doing this, lets update the `_write_to_csv` to execute the body of the method within a try catch and make sure we log an error if we catch an exception. I think `DictWriter` will return an exception if we try to write rows with unexpected fields. We can swallow the exception and let the system continue but we should at least log the error.

Also update all other daos to do the same.